src/background/tab-state.js

```diff
@@ -140,15 +143,47 @@ export default function TabState(initialState, onchange) {
   };
 
   /**
-   * Request the current annotation count for the tab's URL
+   * Request the current annotation count for the tab's URL.
+   * It debounces previous requests within the value defined by
+   * WAITMILLISECONDS
    *
    * @method
    * @param {integer} tabId The id of the tab.
    * @param {string} tabUrl The URL of the tab.
    */
   this.updateAnnotationCount = function (tabId, tabUrl) {
```

Comment:
The combination of `Promise`s and `async/await` here makes this method's flow a little hard to follow. Perhaps choose one?

One thing that would help would be to document this method's return type. e.g. it returns a `Promise` resolving to...?